ddtrace/context.py

```diff
@@ -17,50 +20,40 @@
 log = get_logger(__name__)
 
 
+@attr.s(eq=True, slots=True)
 class Context(object):
-    """Represents the state required to propagate a trace across process
+    """Represents the state required to propagate a trace across execution
     boundaries.
     """
 
-    __slots__ = ["trace_id", "span_id", "_lock", "_span", "_meta", "_metrics"]
-
-    def __init__(
-        self,
-        trace_id=None,  # type: Optional[int]
-        span_id=None,  # type: Optional[int]
-        sampling_priority=None,  # type: Optional[NumericType]
-        dd_origin=None,  # type: Optional[str]
-    ):
-        # type: (...) -> None
-        self.trace_id = trace_id
-        self.span_id = span_id
-
-        self._lock = threading.RLock()
-        self._meta = {}  # type: _MetaDictType
-        self._metrics = {}  # type: _MetricDictType
-        self.dd_origin = dd_origin
-        self.sampling_priority = sampling_priority
-
-    def __eq__(self, other):
-        return (
-            self.span_id == other.span_id
-            and self.trace_id == other.trace_id
-            and self.sampling_priority == other.sampling_priority
-            and self.dd_origin == other.dd_origin
-            and self._meta == other._meta
-            and self._metrics == other._metrics
-        )
+    trace_id = attr.ib(default=None)  # type: Optional[int]
```

Comment:
```suggestion
    trace_id = attr.ib(default=None, type=Optional[int])
```

? Is this a change we can/want to make?